### PR TITLE
network: bridge: add support for  IFLA_BR_FDB_MAX_LEARNED 

### PR DESCRIPTION
**Pull request by tq-gregorHerburger**
Since Linux commit ddd1ad68826d ("net: bridge: Add netlink knobs for number
 / max learned FDB entries") [1] it is possible to limit to number of dynamically learned
fdb entries per bridge.

Add support to the systemd for the IFLA_BR_FDB_MAX_LEARNED and IFLA_BR_N_LEARNED 
netlink attributes.

[1] https://lore.kernel.org/all/20231016-fdb_limit-v5-0-32cddff87758@avm.de/

### CODE REVIEW

File: src/network/netdev/bridge.c
Comment (yuwata) [inline src/network/netdev/bridge.c]: Please drop the error message, like the other options in the above, and simply propagate the error code: `return r;`

File: src/network/netdev/netdev-gperf.gperf
Comment (yuwata) [inline src/network/netdev/netdev-gperf.gperf]: Missing documentation for the setting. Please update man/systemd.netdev.xml .

**Review by yuwata — CHANGES_REQUESTED** [review]

File: src/network/netdev/bridge.c
Comment (tq-gregorHerburger) [reply src/network/netdev/bridge.c]: Ok. Changed it.

File: src/network/netdev/netdev-gperf.gperf
Comment (tq-gregorHerburger) [reply src/network/netdev/netdev-gperf.gperf]: Ok. Added

**Comment by gregorHerburger** [comment]
The mkosi Pipline tests fail for distros where iproute2 is lower than v6.6.0 because they miss this commit: https://github.com/iproute2/iproute2/commit/48cb4320487aa29139537adaca675e1ac6c44ec6

File: test/test-network/systemd-networkd-tests.py
Comment (yuwata) [inline test/test-network/systemd-networkd-tests.py]: > The mkosi Pipline tests fail for distros where iproute2 is lower than v6.6.0 because they miss this commit: [iproute2/iproute2@48cb432](https://github.com/iproute2/iproute2/commit/48cb4320487aa29139537adaca675e1ac6c44ec6)

Then, use something like the following:
```python
if 'fdb_max_learned' in output:
    self.assertIn('fdb_max_learned 4 ', output)
```

File: test/test-network/systemd-networkd-tests.py
Comment (yuwata) [inline test/test-network/systemd-networkd-tests.py]: Also here.

File: src/network/networkctl.c
Comment (yuwata) [inline src/network/networkctl.c:419]: Please introduce a flag, say `bool has_fdb_learned;`

File: src/network/networkctl.c
Comment (yuwata) [inline src/network/networkctl.c]: Then,
```c
if (sd_netlink_message_read_u32(m, IFLA_BR_FDB_MAX_LEARNED, &info->fdb_max_learned) >= 0 &&
    sd_netlink_message_read_u32(m, IFLA_BR_FDB_N_LEARNED, &info->fdb_n_learned) >= 0)
        info->has_fdb_learned = true;
```

File: src/network/networkctl.c
Comment (yuwata) [inline src/network/networkctl.c]: And conditionalize the entries with the `has_fdb_learned` flag.

File: src/network/netdev/bridge.c
Comment (yuwata) [inline src/network/netdev/bridge.c]: So, we do not pass `0`, that contradict with the document.
Maybe, better to introduce a flag, say, `Bridge.fdb_max_learned_set`, and pass `IFLA_BR_FDB_MAX_LEARNED` only when the flag is set.

File: src/network/netdev/netdev-gperf.gperf
Comment (yuwata) [inline src/network/netdev/netdev-gperf.gperf]: Then, need to introduce a custom conf parser.

**Review by yuwata — CHANGES_REQUESTED** [review]

File: test/test-network/systemd-networkd-tests.py
Comment (tq-gregorHerburger) [reply test/test-network/systemd-networkd-tests.py]: Thanks, added.

File: test/test-network/systemd-networkd-tests.py
Comment (tq-gregorHerburger) [reply test/test-network/systemd-networkd-tests.py]: Thanks, added.

File: src/network/networkctl.c
Comment (tq-gregorHerburger) [reply src/network/networkctl.c:419]: Ok. Added.

File: src/network/networkctl.c
Comment (tq-gregorHerburger) [reply src/network/networkctl.c]: Thanks, added.

File: src/network/netdev/bridge.c
Comment (tq-gregorHerburger) [reply src/network/netdev/bridge.c]: Ah I see. Updated.

File: src/network/netdev/bridge.c
Comment (yuwata) [inline src/network/netdev/bridge.c]: Please drop the empty line. We usually write error handling soon after function call.

File: src/network/netdev/bridge.c
Comment (yuwata) [inline src/network/netdev/bridge.c]: Please reset only when an empty string is passed. Otherwise, 
```
FDBMaxLearned=42
FDBMaxLearned=foo
```
will disable the first assignment. We usually ignore all invalid lines, but keep the previous valid assignments. So,
```c
if (isempty(rvalue)) {
        b->fdb_max_learned_set = false;
        return 0;
}
```

**Review by yuwata — CHANGES_REQUESTED** [review]
Looks mostly good. Several minor requests.

**Comment by gregorHerburger** [comment]
Is there anything needed from my side to progress with this PR?

**Review by bluca — APPROVED** [review]

**Comment by bluca** [comment]
> Is there anything needed from my side to progress with this PR?

Sorry for the delay, I clicked the rebase button as the CI last run a month ago, but once it's green I'll merge

File: src/network/netdev/bridge.c
Comment (YHNdnzj) [inline src/network/netdev/bridge.c:249]:
```suggestion
                void *userdata) {

```

**Comment by bluca** [comment]
whitespace change, no need to wait for full ci

**Pull request merged**
